engine/src/main/java/org/terasology/engine/core/GameScheduler.java

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.core;
+
+import org.terasology.gestalt.module.sandbox.API;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+
+@API
+public class GameScheduler {
+
+     private static final Scheduler MAIN;
+
+    static {
+        MAIN = Schedulers.fromExecutor(runnable -> GameThread.asynch(runnable));
+    }
+
+    public static Scheduler gameMain() {
+        return  MAIN;
+    }
+
+    public static Scheduler boundedElastic() {
+        return AccessController.doPrivileged((PrivilegedAction<Scheduler>) Schedulers::boundedElastic);
```

Comment:
A key point for my understanding is that these all return _singletons._ They are _not_ factory methods that return a new Scheduler each time. Makes more sense now.